src/sentry/static/sentry/app/views/organizationEventsV2/eventDetails.jsx

```diff
@@ -2,25 +2,70 @@ import React from 'react';
 import styled from 'react-emotion';
 import {browserHistory} from 'react-router';
 import PropTypes from 'prop-types';
+import {omit} from 'lodash';
 
+import SentryTypes from 'app/sentryTypes';
 import AsyncComponent from 'app/components/asyncComponent';
 import InlineSvg from 'app/components/inlineSvg';
 import withApi from 'app/utils/withApi';
 import space from 'app/styles/space';
 
 import EventModalContent from './eventModalContent';
+import {getQuery} from './utils';
+
+const slugValidator = function(props, propName, componentName) {
+  const value = props[propName];
+  if (value && !/^(?:[^:]+:)?(?:[^:]+):(?:[a-f0-9]+|latest|oldest)$/.test(value)) {
+    return new Error(`Invalid value for ${propName} provided to ${componentName}.`);
+  }
+  return null;
+};
 
 class EventDetails extends AsyncComponent {
   static propTypes = {
-    params: PropTypes.object,
-    eventSlug: PropTypes.string.isRequired,
+    organization: SentryTypes.Organization.isRequired,
+    eventSlug: slugValidator,
```

Comment:
Do we want to validate these at runtime as well? (I guess receiving a 404 makes sense)

Reply:
I think the 404 we display in the modal could use some work, but I can do that separately.